apache/superset
# chore: Upgrade Webpack to v5

### SUMMARY
What's been done:

- Bump webpack from 4.42.0 to 5.52.1
- Bump webpack-cli, webpack-dev-server, plugins and loaders to latest versions
- Remove redundant plugins and loaders
- Refactor webpack config file according to the migration guide (https://webpack.js.org/migrate/5/) and bumped plugins changelogs
- Replace url-loaders and file-loaders with Asset Modules
- Replace mathjs with math-expression-evaluator. Mathjs was causing an import error, but we've been using it in only 1 component (AnnotationLayer), so I replaced it instead of investigating the error. Pulling an 11MB lib just to parse 1 expression was a waste anyway

Next steps:

- Bump other plugins (especially those marked by dependabot to be security vulnerabilities)
- Attempt to simplify our webpack config file - it's quite complex and some parts of it might be unnecessary

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
no visual changes

### TESTING INSTRUCTIONS
1. Everything should build like before
2. The whole app should work like before - verify that images, fonts, libs are getting loaded properly and there are no runtime errors anywhere
3. Linking superset-ui libraries should work like before

### ADDITIONAL INFORMATION
